src/plugins/ImportKeras/ImportKeras.js

```diff
@@ -0,0 +1,299 @@
+/*globals define*/
+/*eslint-env node, browser*/
+
+/**
+ * Generated by PluginGenerator 2.20.5 from webgme on Tue Sep 10 2019 15:28:36 GMT-0500 (Central Daylight Time).
+ * A plugin that inherits from the PluginBase. To see source code documentation about available
+ * properties and methods visit %host%/docs/source/PluginBase.html.
+ */
+
+define([
+    'plugin/PluginConfig',
+    'text!./metadata.json',
+    'plugin/PluginBase',
+    './utils/JSONModelMaps',
+    './utils/json-model-parser',
+], function (
+    PluginConfig,
+    pluginMetadata,
+    PluginBase,
+    ModelMaps,
+    JSONLayerParser,
+) {
+    'use strict';
+
+    pluginMetadata = JSON.parse(pluginMetadata);
+
+
+    /**
+     * Initializes a new instance of ImportKeras.
+     * @class
+     * @augments {PluginBase}
+     * @classdesc This class represents the plugin ImportKeras.
+     * @constructor
+     */
+    var ImportKeras = function () {
+        // Call base class' constructor.
+        PluginBase.call(this);
+        this.pluginMetadata = pluginMetadata;
+    };
+
+    /**
+     * Metadata associated with the plugin. Contains id, name, version, description, icon, configStructure etc.
+     * This is also available at the instance at this.pluginMetadata.
+     * @type {object}
+     */
+    ImportKeras.metadata = pluginMetadata;
+
+    // Prototypical inheritance from PluginBase.
+    ImportKeras.prototype = Object.create(PluginBase.prototype);
+    ImportKeras.prototype.constructor = ImportKeras;
+
+    /**
+     * Main function for the plugin to execute. This will perform the execution.
+     * Notes:
+     * - Always log with the provided logger.[error,warning,info,debug].
+     * - Do NOT put any user interaction logic UI, etc. inside this method.
+     * - callback always has to be called even if error happened.
+     *
+     * @param {function(Error|null, plugin.PluginResult)} callback - the result callback
+     */
+    ImportKeras.prototype.main = async function(callback){
+        let srcJsonHash = this.getCurrentConfig().srcModel;
+        if (!srcJsonHash) {
+            callback(new Error('Keras Json Not Provided'), this.result);
+            return;
+        }
+        try {
+            this.archName = this.getCurrentConfig().archName;
```

Comment:
It might make more sense to declare this as a regular variable (ie, `let archName = ...`) and simply pass it as an argument to `addNewArchitecture` as it appears to only be used there (it is a little easier to follow that way, too).